### PR TITLE
fix: Graph rename bug fix

### LINKED ISSUE #1594: No graph data when module is renamed
Dissolve lets you rename analysis modules but when they are renamed, the data does not seem to be showing up on the graphs in the module.
I have tried this for HistogramCN and Angle and IntraDistance with the same effect. I run the module first and data is printed on the graph, then I renamed the module, cleared the data and ran the simulation again but this time no data appeared on the graph.
I ran this on IDAaaS.

### PR DESCRIPTION
**Pull request by Danbr4d**
Closes #1594.

### CODE REVIEW

**Comment by Danbr4d** [comment]
Not sure how big an issue is but if you rename to something with a number in then it causes a crash too.

File: src/gui/layerTab.cpp [hunk 1]
```diff
@@ -237,10 +237,13 @@ void LayerTab::moduleNameChanged(const QModelIndex &index, const QString &oldNam
...
     auto *mcw = getControlWidget(module);
     if (mcw)
-        mcw->updateControls();
+        mcw->updateControls(ModuleWidget::RecreateRenderablesFlag);
 
     // Rename processing module data
     dissolve_.processingModuleData().renamePrefix(oldName.toStdString(), newName.toStdString());
+
+    if (mcw)
+        mcw->updateControls();
```
Comment (trisyoungs) [inline src/gui/layerTab.cpp]: If this doesn't help they we may as well remove it.
```suggestion
```

**Review by trisyoungs — APPROVED** [review]
:+1:

**Comment by trisyoungs** [comment]
@Danbr4d the issue with renaming I _think_ boils down to the function which was renaming (badly) the relevant data.  I have updated that particular function in the best way possible.  @rprospero if you wouldn't mind taking a look over the horrible code in question, that'd be great (I'll ask for a review from you). I don't see any other way of achieving the same thing...

**Comment by rprospero** [comment]
@trisyoungs I'm not entirely sure that we need to copy into the other map.  The [cpp reference](https://en.cppreference.com/w/cpp/container/map/extract) seems to indicate that it is a safe action.  The only iterator that is invalidated is the one which is literally being moved.  I think we can do

```
for (auto iter = m.begin(); iter < m.end(); ++m) {
    auto node = m.extract(iter);
    if (test(node.key())) {
        node.key() = new_key(node.key());
        m.insert(std::move(node));
    }
}
```

Even if the iteration invalidation causes an issue, we can still go O(n²) in access.

```
do {
    bool done = true
    for (auto iter = m.begin(); iter < m.end(); ++m) {
        auto node = m.extract(iter);
        if (test(node.key())) {
            node.key() = new_key(node.key());
            m.insert(std::move(node));
        }
        done = false;
        break;
    }
} while(!done)
```

It's still ugly, but at least it's constant memory.

**Comment by trisyoungs** [comment]
@rprospero You're aboslutely right - not sure why I got stuck in the copying methodology. I'm still convinced that there are issues trying to do this in a single loop, so I have reworked in line with your second suggestion, but in a slightly more `std` algorithm way.

**Review by rprospero — APPROVED** [review]

**Pull request merged**